addons/io_hubs_addon/components/utils.py

```diff
@@ -0,0 +1,82 @@
+from . import components_registry
+from .gizmos import update_gizmos
+
+
+def add_component(obj, component_id):
+    component_item = obj.hubs_component_list.items.add()
+    component_item.name = component_id
+
+    component_class = components_registry.get_component_by_name(component_id)
+    if component_class:
+        update_gizmos()
+        for dep_id in component_class.get_deps():
+            dep_class = components_registry.get_component_by_id(dep_id)
+            if dep_class:
+                dep_exists = obj.hubs_component_list.items.find(dep_id) > -1
+                if not dep_exists:
+                    add_component(obj, dep_id)
+            else:
+                print("Dependecy '%s' from module '%s' not registered" %
+                      (dep_id, component_id))
+
+
+def remove_component(obj, component_id):
+    component_items = obj.hubs_component_list.items
+    component_items.remove(component_items.find(component_id))
+    component_class = components_registry.get_component_by_id(component_id)
+    del obj[component_class.get_name()]
+
+    component_class = components_registry.get_component_by_name(component_id)
+    if component_class:
+        update_gizmos()
+        for dep_id in component_class.get_deps():
+            dep_class = components_registry.get_component_by_id(dep_id)
+            dep_id = dep_class.get_id()
+            if dep_class:
+                if not is_dep_required(obj, component_id, dep_id):
```

Comment:
What do you refer to exactly here?

Reply:
Just cleaning up dependent components automatically. I got a slight feeling it might be annoying sometimes. I couldn't actually come up with an example where it would be though, so I think its probably the right thing to do.